uisdk/src/main/java/com/karhoo/uisdk/screen/booking/domain/quotes/KarhooAvailability.kt

```diff
@@ -161,7 +166,28 @@ class KarhooAvailability(private val quotesService: QuotesService, private val a
                                                     .DestinationAddressEvent(null))
     }
 
+    @Suppress("MagicNumber")
+    private fun handleVehicleValidity(vehicles: QuoteList) {
+        val SECONDS_TO_MILLISECONDS = ViewsConstants.VALIDITY_SECONDS_TO_MILLISECONDS_FACTOR
+        val minValidity = 5
+
+        if (vehicles.validity >= minValidity) {
```

Comment:
perhaps we can try 

val refresh_delay = if (vehicles.validity >= minValidity) ViewsConstants.VALIDITY_SECONDS_TO_MILLISECONDS_FACTOR else DEFAULT_INTERVAL

GlobalScope.launch {
                delay(vehicles.validity.times(refresh_delay))
                vehiclesObserver?.let { vehiclesObservable?.subscribe(it) }
            }

Reply:
WOuldn't it be:
delay(refresh_delay)

not 

delay(vehicles.validity.times(refresh_delay))